transaction/std/src/memo/gift_code_funding.rs

```diff
@@ -233,80 +291,107 @@ mod tests {
         ));
 
-        // Check public keys match for successful memo lengths for memos that didn't
-        // error
+        // Check fees are correct for successful memos
+        assert_eq!(memo_len_minus_one.get_fee(), fee);
+        assert_eq!(memo_len_exact.get_fee(), fee);
+
+        // Check public keys are correct for successful memos
         assert!(memo_len_minus_one.public_key_matches(&key));
         assert!(memo_len_exact.public_key_matches(&key));
     }
 
     #[test]
     fn test_gift_code_funding_memo_created_with_overlapping_byte_allocations_fail() {
-        // Create hash bytes and note
-        let note = str::from_utf8(&[b'6'; GiftCodeFundingMemo::NOTE_DATA_LEN - 1]).unwrap();
+        // Create hash bytes, fee bytes, and note
+        let note = str::from_utf8(&[b'6'; GiftCodeFundingMemo::NOTE_DATA_LEN]).unwrap();
         let mut rng: StdRng = SeedableRng::from_seed([1u8; 32]);
         let key = RistrettoPublic::from_random(&mut rng);
+        let fee: u64 = 666;
         let hash_bytes = tx_out_public_key_short_hash(&key);
+        let fee_bytes = fee.to_be_bytes();
 
-        // Purposely overlap memo and public key bytes
+        // Purposely overlap public key, fee, and note bytes
         let mut memo_bytes = [0u8; GiftCodeFundingMemo::MEMO_DATA_LEN];
-        memo_bytes[0..GiftCodeFundingMemo::HASH_DATA_LEN].copy_from_slice(&hash_bytes);
+        memo_bytes[..GiftCodeFundingMemo::HASH_DATA_LEN].copy_from_slice(&hash_bytes);
         memo_bytes
-            [(GiftCodeFundingMemo::HASH_DATA_LEN - 1)..(GiftCodeFundingMemo::MEMO_DATA_LEN - 2)]
-            .copy_from_slice(note.as_bytes());
+            [(GiftCodeFundingMemo::HASH_DATA_LEN - 1)..(GiftCodeFundingMemo::NOTE_OFFSET - 1)]
+            .copy_from_slice(&fee_bytes[1..]);
+        memo_bytes
+            [(GiftCodeFundingMemo::NOTE_OFFSET - 1)..(GiftCodeFundingMemo::MEMO_DATA_LEN - 1)]
+            .copy_from_slice(&[b'6'; GiftCodeFundingMemo::NOTE_DATA_LEN]);
+
         let memo = GiftCodeFundingMemo::from(&memo_bytes);
 
         // Check that the hash isn't correctly verified
         assert!(!memo.public_key_matches(&key));
 
-        // Check that the note is erroneous
+        // Check that fee isn't correct
+        assert_ne!(fee, memo.get_fee());
+
+        // Check that the note isn't correct
         assert_ne!(memo.funding_note().unwrap(), note);
     }
 
     #[test]
-    fn test_gift_code_funding_memo_created_with_corrupted_bytes_fail() {
-        // Initialize note and hash bytes
-        let note = str::from_utf8(&[b'6'; GiftCodeFundingMemo::NOTE_DATA_LEN - 1]).unwrap();
+    fn test_gift_code_funding_memo_fees_close_to_max_fee_process_as_expected() {
+        // Create notes near max length
+        const MAX_FEE: u64 = GiftCodeFundingMemo::MAX_FEE;
+        const MAX_FEE_MINUS_ONE: u64 = GiftCodeFundingMemo::MAX_FEE - 1;
+        const MAX_FEE_PLUS_ONE: u64 = GiftCodeFundingMemo::MAX_FEE + 1;
```

Comment:
DRY: Reuse the previous const, or just use `+ 1`/`- 1` since that's shorter.

```suggestion
        const MAX_FEE_MINUS_ONE: u64 = MAX_FEE - 1;
        const MAX_FEE_PLUS_ONE: u64 = MAX_FEE + 1;
```

Reply:
In going between different memo builders, it's easy to get confused which one you're dealing with, so I chose the longer path